build/git.go

```diff
@@ -48,9 +48,14 @@ func getGitAttributes(ctx context.Context, contextPath string, dockerfilePath st
 		wd, _ = filepath.Abs(filepath.Join(cwd, contextPath))
 	}
 
-	gitc := gitutil.New(gitutil.WithContext(ctx), gitutil.WithWorkingDir(wd))
+	gitc, err := gitutil.New(gitutil.WithContext(ctx), gitutil.WithWorkingDir(wd))
+	if err != nil {
+		logrus.Warnf("Failed to initialize git: %v", err)
```

Comment:
This should only warn if there is a `.git` dir

Reply:
Probably also want to be careful of Git worktree support (where `.git` will exist, but is a file that contains something like `gitdir: /full/path/to/.git/worktrees/foo`).

Reply:
(if this is client-side, it'll "just work" when you run `git` commands but if it's server-side then you want to probably warn that `.git` exists but it's a file or something like that since you won't be able to get the original `.git` directory source in that case)

Reply:
It client-side before sending the build request yes